api/src/main/java/org/openmrs/module/commonreports/reports/FamilyPlanningReportManager.java

```diff
@@ -0,0 +1,177 @@
+package org.openmrs.module.commonreports.reports;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import org.apache.commons.io.IOUtils;
+import org.openmrs.module.commonreports.ActivatedReportManager;
+import org.openmrs.module.initializer.api.InitializerService;
+import org.openmrs.module.reporting.common.MessageUtil;
+import org.openmrs.module.reporting.dataset.definition.SqlDataSetDefinition;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.openmrs.module.reporting.report.ReportDesign;
+import org.openmrs.module.reporting.report.definition.ReportDefinition;
+import org.openmrs.module.reporting.report.manager.ReportManagerUtil;
+import org.openmrs.util.OpenmrsClassLoader;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class FamilyPlanningReportManager extends ActivatedReportManager {
+	
+	@Autowired
+	private InitializerService inizService;
+	
+	@Override
+	public boolean isActivated() {
+		return inizService.getBooleanFromKey("report.familyPlanning.active", false);
+	}
+	
+	@Override
+	public String getVersion() {
+		return "1.0.0-SNAPSHOT";
+	}
+	
+	@Override
+	public String getUuid() {
+		return "efd7ba26-7888-45a8-9184-423833ab79d3";
+	}
+	
+	@Override
+	public String getName() {
+		return MessageUtil.translate("commonreports.report.familyPlanning.reportName");
+	}
+	
+	@Override
+	public String getDescription() {
+		return MessageUtil.translate("commonreports.report.familyPlanning.reportDescription");
+	}
+	
+	private Parameter getStartDateParameter() {
+		return new Parameter("startDate", "Start Date", Date.class);
+	}
+	
+	private Parameter getEndDateParameter() {
+		return new Parameter("endDate", "End Date", Date.class);
+	}
+	
+	private String getSqlString(String resourceName) {
+		
+		InputStream is = null;
+		try {
+			is = OpenmrsClassLoader.getInstance().getResourceAsStream(resourceName);
+			return IOUtils.toString(is, "UTF-8");
+		}
+		catch (Exception e) {
+			throw new IllegalArgumentException("Unable to load resource: " + resourceName, e);
+		}
+		finally {
+			IOUtils.closeQuietly(is);
+		}
+	}
+	
+	@Override
+	public List<Parameter> getParameters() {
+		List<Parameter> params = new ArrayList<Parameter>();
+		params.add(getStartDateParameter());
+		params.add(getEndDateParameter());
+		return params;
+	}
+	
+	@Override
+	public ReportDefinition constructReportDefinition() {
+		
+		ReportDefinition rd = new ReportDefinition();
+		
+		rd.setName(getName());
+		rd.setDescription(getDescription());
+		rd.setParameters(getParameters());
+		rd.setUuid(getUuid());
+		
+		SqlDataSetDefinition sqlDsd = new SqlDataSetDefinition();
+		sqlDsd.setName("Family Planning SQL Dataset");
+		sqlDsd.setDescription("Family Planning SQL Dataset");
```

Comment:
@reagan-meant , could you localise these as well? This should be the last change I guess. Then we will be good to go.

Reply:
Done